pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -1647,7 +1647,7 @@ base type (-1 if base type does not use a typmod). -1 if this type is not a doma
       } else {
         decimalDigits = connection.getTypeInfo().getScale(typeOid, typeMod);
         columnSize = connection.getTypeInfo().getPrecision(typeOid, typeMod);
-        if (columnSize == 0) {
+        if ( sqlType != Types.NUMERIC && columnSize == 0) {
```

Comment:
s/ sqlType/sqlType/

Reply:
I think we need the same logic within `getBestRowIdentifier()` as well. Same field and `columnSize == 0` appears there as well:

```java
      if (columnSize == 0) {
        columnSize = connection.getTypeInfo().getDisplaySize(typeOid, typeMod);
      }
```